<a href="https://colab.research.google.com/github/alexandralima1602/-C8280-./blob/main/actividad1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Descargamos el archivo llamado "data.tsv

In [18]:
!wget /content/data.tsv

/content/data.tsv: Scheme missing.


**Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**

Mostrar la parte superior de los datos (las primeras filas)

In [2]:
!head data.tsv

order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


 parte inferior de los datos (las últimas filas)

In [3]:
!tail data.tsv

1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



RSPTA: Las columnas significan los parametros de toda la data como ID, cantidad, nombre del árticulo, elección_descripción y el precio del articulo, las filas significan los ejemplos de los parametros

**¿Cuántos pedidos parece haber?**

In [11]:
!awk -F'\t' '{ sum += $2 } END { print sum }' data.tsv


4972


Viendo la suma de la cantidad de pedidos llegamos a la conclusiòn de que hay 4972 pedidos

**¿cuantas lineas hay en este archivo?**

In [12]:
!wc -l data.tsv

4624 data.tsv


**¿Qué burrito es más popular, steako chicken?**

In [15]:
!grep -E "Steak Burrito" data.tsv | wc -l


368


In [16]:
!grep -E "Chicken Burrito" data.tsv | wc -l


553


El mas popular es Chiken Burrito por que tiene mayor cantidad de pedidos con 553

**¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?**

lo hice de 2 formas

**forma 1**

In [21]:
!awk -F'\t' '/Chicken Burrito/ && (/Black Beans/ ) { count++ } END { print count }' data.tsv


282


In [22]:
!awk -F'\t' '/Chicken Burrito/ && (/Pinto Beans/ ) { count++ } END { print count }' data.tsv

105


**forma 2**

In [24]:
!echo "Black Beans:"
!grep "Chicken Burrito" data.tsv | grep "Black Beans" | wc -l
!echo "Pinto Beans:"
!grep "Chicken Burrito" data.tsv | grep "Pinto Beans" | wc -l

Black Beans:
282
Pinto Beans:
105


Los burritos de pollo suelen tener frijoles negros como frijoles pintos

**Usa la línea de comandos para descubrir algo interesante sobre los datos dados.**

Para encontrar el ítem más pedido:

In [31]:
!cut -f3 data.tsv | sort | uniq -c | sort -nr | head -n 1

    726 Chicken Bowl
